### DISCUSSION TITLE
Connect to Tor address for LND v1 invoice api

### DISCUSSION THREAD

**Discussion by Godrigos** [raspiblitz/raspiblitz#4043] [Q&A] [answered]
Hi there. I'm new to the hole bitcoin universe and as a node owner too (as you will notice by my question).
I've been searching about it and could not find a way to connect to my v1/invoice api on my raspblitz. I have the invoice macaroon HEX, but don't know exactly to what address I should connect. My node address (from 1ml, amboss, etc) doesn't work, as I expected...
Any tip on how I can access this part of the api through Tor? How did I get this hidden service onion address?

I'm on version 1.8.0 (default LND version).

Ps.: for context sake... I'm trying to use [Nostdress](https://lnaddress.com/lnaddress).

Thanks for the attention and time.

**Comment by rootzoll** [comment, marked as answer]
The LND REST API runs on port 8080 on your RaspiBlitz. So if you call on your browser `https://[LOCALIP]:8080/v1/getinfo` you see some response. To keep in min here:

- you need to use `https` because LND is using their own TLS certificate
- that TLS certificate isa self-signed and you might allow that with your browser
- if you get a response LND will give an error back demanding a macaroon

But you you basically see that the LND REST API is available. 

**But what about Tor?** 

For that we need to get the onion address for the LND REST API first - do get that run when SSH in on command line `sudo cat /mnt/hdd/tor/lndrest/hostname` then you will get back an oniionaddress like for example `7hyube57cxcd47lcgrihujaevtsqlgd5ga34l637rgsyvd32xq6da6id.onion` And you can now take that onien address and use it to test call the API in the Tor browser like with:

`https://7hyube57cxcd47lcgrihujaevtsqlgd5ga34l637rgsyvd32xq6da6id.onion:8080/v1/getinfo`

**Reply by Godrigos** [reply]
Thanks for the response.
Well the good news is that, actually, I was doing it right in my last attempts. But I got and error as response from Nostdress:
`couldn't make an invoice with the given data: call to lnd failed (500): {"code":2,"message":"cannot determine data format of binary-encoded macaroon","details":[]}`.
The site itself ask for HEX macarrons but is not accepting mine. It complains about binary encoding.
Any ideas why or how I can fix that?

**Reply by Godrigos** [reply]
The admin macaroon HEX seems to work, but... Is it safe to use it?

**Reply by rootzoll** [reply]
Using admin macaroons is always a risk - you basically have to trust the system with your funds your using it with. If you want a more fine grained macaroon rights control you have to go into baking custom macaroons yourself with LND: https://docs.lightning.engineering/lightning-network-tools/lnd/macaroons
